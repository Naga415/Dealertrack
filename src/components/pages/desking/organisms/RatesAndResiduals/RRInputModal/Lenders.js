'use strict';
import React from 'react';
import PropTypes from 'prop-types';
import DropdownButton from 'react-bootstrap/lib/DropdownButton';
import MenuItem from 'react-bootstrap/lib/MenuItem';
// eslint-disable-next-line react/prefer-stateless-function
class Lenders extends React.PureComponent {
  constructor(props, context) {
    super(props, context);
  }

  setSelectedLender = (indexStr)=>{
    const index = parseInt(indexStr, 10);
    this.props.setSelectedLender(this.props.lenders[index]);
  }

  render() {
    return (
      <div>
        <DropdownButton
          id='Lenders'
          bsSize="small"
          bsStyle="link"
          title={this.props.selectedLender}>
          {this.props.lenders.map((lender, index)=><MenuItem id={`LenderMenuItem${index}`} onSelect={this.setSelectedLender} key={index} eventKey={`${index}`}>{lender}</MenuItem>)}
        </DropdownButton>
      </div>
    );
  }
}

Lenders.propTypes = {
  lenders:  PropTypes.arrayOf(PropTypes.string),
  selectedLender: PropTypes.string,
  setSelectedLender: PropTypes.func
};

export default Lenders;
