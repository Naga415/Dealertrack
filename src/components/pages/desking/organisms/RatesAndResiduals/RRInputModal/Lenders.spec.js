/**
 * Created by RBANDERSON on 10/4/2017.
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import sinon from 'sinon';
import Lenders from './Lenders';

const umb = 'UMB';
const selectedLender = 'Ally';
const lenders = [umb, selectedLender];
const setSelectedLender = sinon.spy();
const props = {
  lenders,
  selectedLender,
  setSelectedLender
};

function setupmount () {
  return mount(<Lenders {...props} />);
}

describe('Lenders', ()=>{
  it('renders <Lenders/> ', () => {
    const component = renderer.create(<Lenders {...props}/>);
    expect(component.toJSON()).toMatchSnapshot();
  });
  
  it('triggers addSelectedIncentive event when incentive checkbox is clicked', () => {
    const component = setupmount();
    component.find(`#LenderMenuItem${1}`).simulate('click');
    expect(setSelectedLender.calledOnce).toEqual(true);
  });
});
