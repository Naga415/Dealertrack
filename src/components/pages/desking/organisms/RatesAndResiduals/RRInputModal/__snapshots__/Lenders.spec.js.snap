// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lenders renders <Lenders/>  1`] = `
<div>
  <div
    className="dropdown btn-group btn-group-sm btn-group-link"
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle btn btn-sm btn-link"
      disabled={false}
      id="Lenders"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      type="button"
    >
      Ally
       
      <span
        className="caret"
      />
    </button>
    <ul
      aria-labelledby="Lenders"
      className="dropdown-menu"
      role="menu"
    >
      <li
        className=""
        role="presentation"
        style={undefined}
      >
        <a
          href="#"
          id="LenderMenuItem0"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1"
        >
          UMB
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}
      >
        <a
          href="#"
          id="LenderMenuItem1"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1"
        >
          Ally
        </a>
      </li>
    </ul>
  </div>
</div>
`;
